plugin/audit_log/audit_log.c

```diff
@@ -186,16 +187,29 @@ void escape_buf(const char *in, size_t *inlen, char *out, size_t *outlen,
   while ((in < inend) && (out < outend))
   {
     replaced= FALSE;
-    for (rule= escape_rules; rule->character; rule++)
-    {
-      if (*in == rule->character)
-      {
+    if ((unsigned char)(*in) < 32) {
+      if (control_escape_rules[(unsigned int)*in].character) {
```

Comment:
Likewise. If this one is because of sign mismatch, is it possible to make the ```in``` arg ```const unsigned char *``` instead?

Reply:
Because by default char is signed in gcc. Even if I specify the parameter as unsigned, I have to add this cast somewhere.